Nats server with resolver: {type: full} crashes if `system_account` is not set

## Defect

Make sure that these boxes are checked before submitting your issue -- thank you!

 - [x] Included `nats-server -DV` output
 - [x] Included a [Minimal, Complete, and Verifiable example] (https://stackoverflow.com/help/mcve)

#### Versions of `nats-server` and affected client libraries used:

```sh
$ nats-server -DV
[3662451] 2021/04/24 12:08:54.602759 [INF] Starting nats-server
[3662451] 2021/04/24 12:08:54.602821 [INF]   Version:  2.2.2
[3662451] 2021/04/24 12:08:54.602826 [INF]   Git:      [a5f3aab]
[3662451] 2021/04/24 12:08:54.602830 [DBG]   Go build: go1.16.3
[3662451] 2021/04/24 12:08:54.602834 [INF]   Name:     NCHSX3HTBSRJNWB2IJAM6R2FGBNN5FRZ3X33RGHG23OWHTILXXB7ORLX
[3662451] 2021/04/24 12:08:54.602838 [INF]   ID:       NCHSX3HTBSRJNWB2IJAM6R2FGBNN5FRZ3X33RGHG23OWHTILXXB7ORLX
[3662451] 2021/04/24 12:08:54.602868 [DBG] Created system account: "$SYS"
[3662451] 2021/04/24 12:08:54.603528 [INF] Listening for client connections on 0.0.0.0:4222
[3662451] 2021/04/24 12:08:54.603537 [DBG] Get non local IPs for "0.0.0.0"
[3662451] 2021/04/24 12:08:54.603921 [DBG]   ip=192.168.199.2
[3662451] 2021/04/24 12:08:54.603928 [DBG]   ip=fdfc:6c1c:2d83:10:c1dc:4824:353:6a4a
[3662451] 2021/04/24 12:08:54.603932 [DBG]   ip=fdfc:6c1c:2d83:10:81f2:d9e:85d:6b40
[3662451] 2021/04/24 12:08:54.603936 [DBG]   ip=fdfc:6c1c:2d83:10:c416:6500:883c:682c
[3662451] 2021/04/24 12:08:54.603965 [DBG]   ip=192.168.122.1
[3662451] 2021/04/24 12:08:54.604022 [DBG]   ip=192.168.39.1
[3662451] 2021/04/24 12:08:54.604076 [DBG]   ip=172.19.0.1
[3662451] 2021/04/24 12:08:54.604116 [DBG]   ip=172.17.0.1
[3662451] 2021/04/24 12:08:54.604155 [INF] Server is ready
^C[3662451] 2021/04/24 12:08:57.148228 [DBG] Trapped "interrupt" signal
[3662451] 2021/04/24 12:08:57.148324 [INF] Initiating Shutdown...
[3662451] 2021/04/24 12:08:57.148462 [DBG] SYSTEM - System connection closed: Client Closed
[3662451] 2021/04/24 12:08:57.148471 [DBG] Client accept loop exiting..
[3662451] 2021/04/24 12:08:57.148503 [INF] Server Exiting..
```

#### OS/Container environment:

Ubuntu 20.04.2 LTS

#### Steps or code to reproduce the issue:

Use this config

```
trace: true
operator: /home/sindre/.nsc/nats/O_operator/O_operator.jwt
resolver: {
    type: full
    dir: '/home/sindre/src/iterate/app.iterate.no/nats/testing_nsc/jwt'
}
```

Start nats-server

```
nats-server -c nas.config
```

#### Expected result:

Error message: `system_account required` or at least not crashing

#### Actual result:

Server crashes with a SIGSEGV

```
$ nats-server -c nas.config
[3663003] 2021/04/24 12:10:15.199546 [INF] Starting nats-server
[3663003] 2021/04/24 12:10:15.199613 [INF]   Version:  2.2.2
[3663003] 2021/04/24 12:10:15.199618 [INF]   Git:      [a5f3aab]
[3663003] 2021/04/24 12:10:15.199627 [INF]   Name:     NDREZQBDOGGJJFX24Z5JMY3J6PC2QKF5PCP64ZTSPQHXEX667D3WVSA3
[3663003] 2021/04/24 12:10:15.199631 [INF]   ID:       NDREZQBDOGGJJFX24Z5JMY3J6PC2QKF5PCP64ZTSPQHXEX667D3WVSA3
[3663003] 2021/04/24 12:10:15.199640 [INF] Using configuration file: nas.config
[3663003] 2021/04/24 12:10:15.199645 [INF] Trusted Operators
[3663003] 2021/04/24 12:10:15.199651 [INF]   System  : ""
[3663003] 2021/04/24 12:10:15.199656 [INF]   Operator: "O_operator"
[3663003] 2021/04/24 12:10:15.199677 [INF]   Issued  : 2021-04-24 11:41:08 +0200 CEST
[3663003] 2021/04/24 12:10:15.199683 [INF]   Expires : 1970-01-01 01:00:00 +0100 CET
[3663003] 2021/04/24 12:10:15.199690 [WRN] Trusted Operators should utilize a System Account
[3663003] 2021/04/24 12:10:15.199761 [INF] Server is ready
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xa0 pc=0x7e467a]

goroutine 1 [running]:
github.com/nats-io/nats-server/server.(*Server).newRespInbox(0xc0001b2000, 0xc000144150, 0x38)
	/home/travis/gopath/src/github.com/nats-io/nats-server/server/events.go:1705 +0x3a
github.com/nats-io/nats-server/server.(*DirAccResolver).Start(0xc00007eb60, 0xc0001b2000, 0x0, 0x0)
	/home/travis/gopath/src/github.com/nats-io/nats-server/server/accounts.go:3634 +0x1d9
github.com/nats-io/nats-server/server.(*Server).Start(0xc0001b2000)
	/home/travis/gopath/src/github.com/nats-io/nats-server/server/server.go:1524 +0xf78
github.com/nats-io/nats-server/server.Run(...)
	/home/travis/gopath/src/github.com/nats-io/nats-server/server/service.go:21
main.main()
	/home/travis/gopath/src/github.com/nats-io/nats-server/main.go:116 +0x188
```

--- a/server/jwt.go
+++ b/server/jwt.go
@@ -114,10 +114,9 @@ func validateTrustedOperators(o *Options) error {
 	} else if o.TrustedOperators[0].SystemAccount == _EMPTY_ {
 		// In case the system account is neither defined in config nor in the first operator.
 		// If it would be needed due to the nats account resolver, raise an error.
-		_, okDir := o.AccountResolver.(*DirAccResolver)
-		_, okCAcheDir := o.AccountResolver.(*CacheDirAccResolver)
-		if okDir || okCAcheDir {
-			return fmt.Errorf("using nats based account resolver - the system account needs to be specified in config or the operator jwt")
+		switch o.AccountResolver.(type) {
+		case *DirAccResolver, *CacheDirAccResolver:
+			return fmt.Errorf("using nats based account resolver - the system account needs to be specified in configuration or the operator jwt")
 		}
 	}
 	ver := strings.Split(strings.Split(VERSION, "-")[0], ".RC")[0]

--- a/server/jwt.go
+++ b/server/jwt.go
@@ -96,11 +96,11 @@ func validateTrustedOperators(o *Options) error {
 	if len(o.TrustedOperators) > 0 && len(o.TrustedKeys) > 0 {
 		return fmt.Errorf("conflicting options for 'TrustedKeys' and 'TrustedOperators'")
 	}
-	if o.SystemAccount != "" {
+	if o.SystemAccount != _EMPTY_ {
 		foundSys := false
 		foundNonEmpty := false
 		for _, op := range o.TrustedOperators {
-			if op.SystemAccount != "" {
+			if op.SystemAccount != _EMPTY_ {
 				foundNonEmpty = true
 			}
 			if op.SystemAccount == o.SystemAccount {
@@ -110,6 +110,14 @@ func validateTrustedOperators(o *Options) error {
 		}
 		if foundNonEmpty && !foundSys {
 			return fmt.Errorf("system_account in config and operator JWT must be identical")
+		}
+	} else if o.TrustedOperators[0].SystemAccount == _EMPTY_ {
+		// In case the system account is neither defined in config nor in the first operator.
+		// If it would be needed due to the nats account resolver, raise an error.
+		_, okDir := o.AccountResolver.(*DirAccResolver)
+		_, okCAcheDir := o.AccountResolver.(*CacheDirAccResolver)
+		if okDir || okCAcheDir {
+			return fmt.Errorf("using nats based account resolver - the system account needs to be specified in config or the operator jwt")
 		}
 	}
 	ver := strings.Split(strings.Split(VERSION, "-")[0], ".RC")[0]

--- a/server/jwt_test.go
+++ b/server/jwt_test.go
@@ -5757,6 +5757,7 @@ func TestJWTNoSystemAccountButNatsResolver(t *testing.T) {
 			s, err := NewServer(opts)
 			s.Shutdown()
 			require_Error(t, err)
+			require_Contains(t, err.Error(), "the system account needs to be specified in configuration or the operator jwt")
 		})
 	}
 }

--- a/server/jwt_test.go
+++ b/server/jwt_test.go
@@ -5739,3 +5739,24 @@ func TestJWTMappings(t *testing.T) {
 	require_Len(t, 1, updateJwt(t, srv.ClientURL(), sysCreds, aJwtNoM, 1))
 	test("foo2", "bar2", true)
 }
+
+func TestJWTNoSystemAccountButNatsResolver(t *testing.T) {
+	dirSrv := createDir(t, "srv")
+	defer removeDir(t, dirSrv)
+	for _, resType := range []string{"full", "cache"} {
+		t.Run(resType, func(t *testing.T) {
+			conf := createConfFile(t, []byte(fmt.Sprintf(`
+			listen: -1
+			operator: %s
+			resolver: {
+				type: %s
+				dir: %s
+			}`, ojwt, resType, dirSrv)))
+			defer removeFile(t, conf)
+			opts := LoadConfig(conf)
+			s, err := NewServer(opts)
+			s.Shutdown()
+			require_Error(t, err)
+		})
+	}
+}

--- a/server/leafnode_test.go
+++ b/server/leafnode_test.go
@@ -28,6 +28,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/nats-io/nkeys"
+
 	jwt "github.com/nats-io/jwt/v2"
 	"github.com/nats-io/nats.go"
 )
@@ -2402,6 +2404,10 @@ func TestLeafNodeRouteParseLSUnsub(t *testing.T) {
 }
 
 func TestLeafNodeOperatorBadCfg(t *testing.T) {
+	sysAcc, err := nkeys.CreateAccount()
+	require_NoError(t, err)
+	sysAccPk, err := sysAcc.PublicKey()
+	require_NoError(t, err)
 	tmpDir := createDir(t, "_nats-server")
 	defer removeDir(t, tmpDir)
 	for errorText, cfg := range map[string]string{
@@ -2421,6 +2427,7 @@ func TestLeafNodeOperatorBadCfg(t *testing.T) {
 			conf := createConfFile(t, []byte(fmt.Sprintf(`
 		port: -1
 		operator: %s
+		system_account: %s
 		resolver: {
 			type: cache
 			dir: %s
@@ -2428,7 +2435,7 @@ func TestLeafNodeOperatorBadCfg(t *testing.T) {
 		leafnodes: {
 			%s
 		}
-	`, ojwt, tmpDir, cfg)))
+	`, ojwt, sysAccPk, tmpDir, cfg)))
 			defer removeFile(t, conf)
 			opts := LoadConfig(conf)
 			s, err := NewServer(opts)

--- a/server/server.go
+++ b/server/server.go
@@ -1664,6 +1664,9 @@ func (s *Server) Start() {
 // Shutdown will shutdown the server instance by kicking out the AcceptLoop
 // and closing all associated clients.
 func (s *Server) Shutdown() {
+	if s == nil {
+		return
+	}
 	// Transfer off any raft nodes that we are a leader by shutting them all down.
 	s.shutdownRaftNodes()
 

